save-core/src/commonMain/kotlin/org/cqfn/save/core/utils/ProcessBuilder.kt

```diff
@@ -21,6 +25,103 @@ expect class ProcessBuilder() {
     fun exec(command: List<String>, redirectTo: Path?): ExecutionResult
 }
 
+/**
+ * Class contains common fields for all platforms
+ */
+class ProcessBuilderInternal {
+    /**
+     * Typealias
+     */
+    val fs = FileSystem.SYSTEM
+
+    /**
+     * Temporary directory for stderr and stdout (popen can't separate streams, so we do it ourselves)
+     */
+    val tmpDir = (FileSystem.SYSTEM_TEMPORARY_DIRECTORY / this::class.simpleName!!).also {
+        fs.createDirectory(it)
+    }
+
+    /**
+     * Path to stdout file
+     */
+    val stdoutFile = tmpDir / "stdout.txt".also { logDebug("Created file for stdout of ProcessBuilder in: $tmpDir") }
+
+    /**
+     * Path to stderr file
+     */
+    val stderrFile = tmpDir / "stderr.txt".also { logDebug("Created file for stderr of ProcessBuilder in: $tmpDir") }
+
+    /**
+     *  Read data from stdout file, we will use it in [ExecutionResult]
+     *
+     * @return string containing stdout
+     */
+    fun getStdout(): List<String> {
+        val stdout = fs.read(stdoutFile) {
+            generateSequence { readUtf8Line() }.toList()
+        }
+        return stdout
+    }
+
+    /**
+     * Read data from stderr file, we will use it in [ExecutionResult]
+     *
+     * @return string containing stderr
+     */
+    fun getStderr(): List<String> {
+        val stderr = fs.read(stderrFile) {
+            generateSequence { readUtf8Line() }.toList()
+        }
+        return stderr
+    }
+
+    /**
+     * Modify execution command for popen,
+     * stderr will be redirected to tmp file
+     *
+     * @param command raw command
+     * @return command with redirection of stderr to tmp file
+     */
+    fun prepareCmd(command: List<String>): String {
+        val cmd = command.joinToString(" ") + " 2>$stderrFile"
```

Comment:
What if command already contains stderr redirection? I think we should at least warn the user in this case.

Reply:
bash allows only one redirection per file descriptor.
If multiple redirections will be provided, they will processed from left to right, and the last one will take effect.
So we can warn the user, and don't really care about his redirection, since only our will be applied.